textx/generators.py

```diff
@@ -1,58 +1,83 @@
 import click
 import os
-from textx import generator
+from functools import partial
+from textx.registration import generator
 from textx.export import metamodel_export, model_export, PlantUmlRenderer
 
 
-@generator('textX', 'dot')
-def metamodel_generate_dot(metamodel, model, output_path, overwrite, debug):
-    "Generating dot visualizations from textX grammars"
+def get_output_file(input_file, output_path, fileext):
```

Comment:
I would suggest to call the function `get_output_filename` or `create_output_filename`, because it is not clear if a file get returned or a filename.

Reply:
@goto40 Good catch! Thanks.